enip_backend/calls_gsheet_sync/run.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+import psycopg2
+from pytz import timezone
+
+from ..enip_common.config import CALLS_GSHEET_ID, POSTGRES_URL
+from ..enip_common.gsheets import get_gsheets_client, get_worksheet_data
+
+DATA_RANGE = ("A", "D")
+EXPECTED_DATA_HEADER = ["State", "AP Call", "AP Called At (ET)", "Published?"]
+
+PUBLISH_DEFAULT_RANGE = ("F1", "G1")
+
+CALLED_AT_TZ = timezone("US/Eastern")
+CALLED_AT_FMT = "%m/%d/%Y %H:%M:%S"
+
+
+def _update_calls_sheet_from_db(cursor, table, worksheet):
+    cursor.execute(
+        f"SELECT state, ap_call, ap_called_at FROM {table} WHERE ap_call IS NOT NULL"
+    )
+    db_calls_lookup = {r[0]: {"call": r[1], "called_at": r[2]} for r in cursor}
+    sheet_data = get_worksheet_data(worksheet, DATA_RANGE, EXPECTED_DATA_HEADER)
+    # Selectively update cells that don't have the same api calls from the db
+    # Note: we intentionally don't insert new rows into the sheet for states
+    # that exist in the db but not the sheet. The sheet will need to be seeded
+    # with the states we want to report on.
```

Comment:
ahhh I was going to ask if this is a scenario that we're hoping never happens -- but I guess this is the mechanism by which we're filtering out House races? 👍

Reply:
i wrote it this way because i thought the test data was from a different election. In any case this is simpler because we don't have to insert or reorder rows and, from the perspective of this sync, some columns are writable and some are read-only. If we had to manage a long or changing list list of races I would definitely rewrite it to insert.